test/signer_test.exs

```diff
@@ -130,4 +133,19 @@ defmodule AntikytheraAws.Signer.V4Test do
     path_with_qs = "/hoge?foo=bar"
     catch_error V4.prepare_headers(@example_creds, @example_region, "iam", :get, path_with_qs, "", %{"host" => "example.com"}, [{"foo", "bar"}])
   end
+
+  test "prepare_headers/8 should generate 'x-amz-content-sha256' if service is S3 and method is POST or PUT" do
+    result = V4.prepare_headers(@example_creds, @example_region, "s3", :put, "/object_key", @example_payload, %{"host" => "example.com"}, [])
+    assert result["x-amz-content-sha256"] == hex_sha256(@example_payload)
+  end
+
+  test "prepare_headers/8 should not generate 'x-amz-content-sha256' if service is S3 and method is other than POST and PUT" do
+    result = V4.prepare_headers(@example_creds, @example_region, "s3", :get, "/", "", %{"host" => "example.com"}, [])
```

Comment:
Let us use "/object_key" path here too, in order to emphasize difference from the above clause.

Reply:
fixed in 9dd183d